CardGenerator.py

```diff
@@ -465,22 +469,20 @@ def _draw_back(self, canvas):
         custom_scale = min(1.0, 20 / len(self.title))
         canvas.setFillColor("black")
         title_font_height = self.fonts.set_font(canvas, "title", custom_scale)
-        title_line_bottom = (
-            self.height - self.border_back[Border.TOP] - self.TITLE_BAR_HEIGHT
-        )
+        title_line_bottom = self.frames[0]._y2 + self.STANDARD_BORDER
         title_bottom = (
             title_line_bottom + (self.TITLE_BAR_HEIGHT - title_font_height) / 2
         )
-        canvas.drawCentredString(
-            self.width + self.BASE_WIDTH / 2, title_bottom, self.title.upper()
-        )
+        title_center = (self.frames[0]._x1 + self.frames[0]._x2) / 2
```

Comment:
How come these calculations have changed to be based on  the size of the frames?

Reply:
I failed to wrap my head around the size and bleed thing. I'll try once more, but the formula may be wrong again:

Assume b is bleed and B is border, w is base width and W is width, d is divider, W = 2w+2b, and the center of any text is between the left and right frame bounds, so between W+B and 1.5W-0.5d, which is 0.5(W+B+1.5W-0.5d) = (1.25W+0.5B-0.25d) = (1W+0.25W+0.5B-0.25d) = (1W+0.5w+0.5b+0.5B-0.25d). Using base width alone was wrong, the formula above would have been just (1W+0.5w).

Alternatively, the text center is right in the middle of x1 and x2 of the frame.

I noticed because the text started being just a tad too much on the right at first. If you give it 10mm bleed, it was much more pronounced.